string_unique/string-app/src/test/java/net/mtnboy/StringUnique/AppTest.java

```diff
@@ -28,20 +28,46 @@ public static Test suite()
         return new TestSuite( AppTest.class );
     }
 
-    public void testApp()
+    public void testEmpties()
     {
-        assertEquals(App.hasAllUnique(null), true);
-        assertEquals(App.hasAllUnique(""), true);
+        assertTrue("check null", App.hasAllUnique(null));
+        assertTrue("check empty", App.hasAllUnique(""));
+    }
 
-        assertEquals(App.hasAllUnique("01234{}"), true);
-        assertEquals(App.hasAllUnique("01234{}0"), false);
+    public void testNums() {
+        assertTrue("check nums",  App.hasAllUnique("01234{}"));
+        assertFalse("check nums with dup",  App.hasAllUnique("01234{}0"));
+    }
 
-        assertEquals(App.hasAllUnique("abcABC"), true);
-        assertEquals(App.hasAllUnique("ABCABC"), false);
+    public void testCase() {
+        assertTrue("check mixed case", App.hasAllUnique("abcABC"));
+        assertFalse("check upcase dups", App.hasAllUnique("ABCABC"));
+    }
 
-        //skip long string test because I'm lazy to build a unit test for this. TBD with another test framework.
+    //skip long string test because I'm lazy to build a unit test for this. TBD with another test framework.
+
+    public void testShort() {
+        assertTrue("check singleton", App.hasAllUnique("a"));
+        assertFalse("check dup, but short", App.hasAllUnique("aa"));
+    }
+
+    public void testDup() {
+        assertFalse(App.hasAllUnique("hello"));
+        assertFalse(App.hasAllUnique("heloh"));
+        assertFalse(App.hasAllUnique("eehlo"));
+        assertFalse(App.hasAllUnique("ehloh"));
+    }
+
+    public void testNoDup() {
+        assertTrue("simple no dup", App.hasAllUnique("abcdefghijklmno"));
+        assertTrue("upcase no dup", App.hasAllUnique("ABCDEFG"));
+        assertTrue("mixcase, no dup", App.hasAllUnique("abcCBA"));
+    }
 
-        assertEquals(App.hasAllUnique("a"), true);
-        assertEquals(App.hasAllUnique("aa"), false);
+    public void testWhiteSpace() {
+        assertTrue("white space, no dup", App.hasAllUnique(" "));
+        assertFalse("double space, dup", App.hasAllUnique("  "));
+        assertTrue("mixed space, no dup", App.hasAllUnique(" \t"));
+        assertTrue("mixed space no dup2", App.hasAllUnique(" \n"));
     }
```

Comment:
Special characters test?

Reply:
added below. Not trivial in java. I'm assuming you meant extended.